sel/src/Sel/Internal.hs

```diff
@@ -33,3 +35,20 @@ foreignPtrOrd fptr1 fptr2 size =
               | result == 0 -> EQ
               | result < 0 -> LT
               | otherwise -> GT
+
+-- | Securely allocate an amount of memory with 'sodiumMalloc' and pass
+-- a pointer to the region to the provided action.
+-- The region is deallocated with 'sodiumFree' afterwards.
+-- Do not try to jailbreak the pointer outside of the action,
+-- this will not be pleasant.
+allocateWith
+  :: CSize
+  -- ^ Amount of memory to allocate
+  -> (Ptr a -> IO b)
```

Comment:
Should we have MonadIO m instead of IO? @kozross

Reply:
Hmm. There's some reason to be general with a callback argument, as it would otherwise require unlifts. I'd say that'd be fine, something like:

``allocateWith :: forall (m :: Type -> Type) (a :: Type) . MonadIO m => (Ptr a -> m b) -> m b``